library/alloc/src/rc.rs

```diff
@@ -386,7 +386,35 @@ impl<T> Rc<T> {
     /// returns, calling [`upgrade`] on the weak reference inside `data_fn` will
     /// fail and result in a `None` value.
     ///
+    /// # Why using this method?
+    ///
+    /// It allows to construct a `T` which holds a `Weak<T>` pointer to itself.
+    /// Without this method, you would need to use [`Weak::new`] and then
+    /// mutate the value by using [`Rc::get_mut`] to fix up the self references.
+    ///
+    /// # Why is a closure needed?
```

Comment:
Things like "Without this .." and "Why is a closure needed?" makes it seem like this is part of an RFC that is arguing for the addition of this function. But as part of the documentation of this (stable) function, I'm not sure if it helps to add comparisions with a world without this function.